src/theory/builtin/theory_builtin_rewriter.h

```diff
@@ -82,12 +56,34 @@ class TheoryBuiltinRewriter {
 
 // conversions between lambdas and arrays
 private:  
+  /** recursive helper for getLambdaForArrayRepresentation */
   static Node getLambdaForArrayRepresentationRec( TNode a, TNode bvl, unsigned bvlIndex, 
                                                   std::unordered_map< TNode, Node, TNodeHashFunction >& visited );
 public:
-  /** given an array constant a, returns a lambda expression that it corresponds to, with bound variable list bvl. */
+  /** 
+   * Given an array constant a, returns a lambda expression that it corresponds to, with bound variable list bvl. 
+   * Examples:
+   *
+   * (store 0 1 (storeall (Array Int Int) 2)) 
+   * becomes
+   * ((lambda x. (ite (= x 0) 1 2))
+   *
+   * (store 0 (store 1 2 (storeall (Array Int Int) 3)) (storeall (Array Int (Array Int Int)) (storeall (Array Int Int) 4)))
+   * becomes
+   * (lambda xy. (ite (= x 0) (ite (= x 1) 2 3) 4))
+   *
+   * (store 1 true (store 2 true (storeall (Array Int Bool) false)))
+   * becomes
+   * (lambda x. (ite (= x 1) true (ite (= x 2) true false)))
+   *
+   * Notice that the return body of the lambda is rewritten to ensure that the representation is canonical. Hence the last
+   * example will in fact be returned as:
+   * (lambda x. (ite (= x 1) true (= x 2)))
+   */
   static Node getLambdaForArrayRepresentation( TNode a, TNode bvl );
```

Comment:
Nice examples!  This helps a lot.  One minor thing: in SMT-LIB, the order of arguments for store is array index value.  It looks like you've done index value array here.  Can you rewrite to follow the standard?